### PR TITLE
TGP-457 - Create homepage

### PR DESCRIPTION
**Pull request by ChrisCookOC**

### CODE REVIEW

File: a11y/FrontendAccessibilitySpec.scala [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import models.{CheckMode, Mode, NormalMode}
+import models.{Mode, NormalMode}
```
Comment (Ewan-James-Donovan) [inline a11y/FrontendAccessibilitySpec.scala]: Do we want accessibility for check mode, too? Maybe not a question for this ticket.

File: app/controllers/CheckYourAnswersController.scala [hunk 2]
```diff
@@ -43,9 +46,14 @@ class CheckYourAnswersController @Inject() (
...
-  // TODO replace dummy route and post session data
-  def onSubmit: Action[AnyContent] = authorise { implicit request =>
-    Redirect(routes.DummyController.onPageLoad.url)
+  def onSubmit: Action[AnyContent] = (authorise andThen getData).async { implicit request =>
+    routerService
+      .setUpProfile(request.eori, request.userAnswers.maintainProfileAnswers)
+      .fold(
+        error => Redirect(routes.DummyController.onPageLoad.url), //TODO redirect to actual error page
+        success => Redirect(routes.HomepageController.onPageLoad.url)
+      )(catsStdInstancesForFuture(defaultExecutionContext))
```
Comment (Ewan-James-Donovan) [inline app/controllers/CheckYourAnswersController.scala]: Is the catsStdInstancesForFuture used for implicit excution context? I'm used to seeing `class XyzClass @Inject() ( <some injects>) (implicit ec ExecutionContext)`. I'm not averse to this cats way if we know it works properly.

File: it/test/base/WireMockServerSpec.scala [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+import com.github.tomakehurst.wiremock.WireMockServer
+import com.github.tomakehurst.wiremock.core.WireMockConfiguration.options
+
+trait WireMockServerSpec {
+
+  implicit lazy val wireMock: WireMockServer = new WireMockServer(options().dynamicPort())
+
+  val wireMockHost: String   = "localhost"
+  lazy val wireMockPort: Int = wireMock.port()
+}
```
Comment (Ewan-James-Donovan) [inline it/test/base/WireMockServerSpec.scala:28]: Love the use of wiremock

**Review by Ewan-James-Donovan — DISMISSED** [review]
Looks good to me, I'll reapprove after conflicts are fixed, no major comments,

File: a11y/FrontendAccessibilitySpec.scala (hunk 1 above)
Comment (ChrisCookOC) [reply a11y/FrontendAccessibilitySpec.scala]: Was wondering that as I removed the import! I think right now the pages are so similar in each mode its not a big concern, but maybe something for down the line

File: app/controllers/CheckYourAnswersController.scala (hunk 2 above)
Comment (ChrisCookOC) [reply app/controllers/CheckYourAnswersController.scala]: nah so it wants an implicit Functor[Future], which is the catsStdInstancesForFuture, which then itself wanted an execution context.

Have moved it up out of the method call and it looks a bit nicer I think. Let me know if that is clearer

File: a11y/FrontendAccessibilitySpec.scala (hunk 1 above)
Comment (Ewan-James-Donovan) [reply a11y/FrontendAccessibilitySpec.scala]: Cool!

File: app/controllers/CheckYourAnswersController.scala (hunk 2 above)
Comment (Ewan-James-Donovan) [reply app/controllers/CheckYourAnswersController.scala]: That's clearer :)

**Review by Ewan-James-Donovan — APPROVED** [review]
Lgtm

**Review by besscerule — APPROVED** [review]

**Comment by besscerule** [comment]
Looks ace :)

**Pull request merged**